packages/NodeCollector/NodeCollector/NodeRepository.php

```diff
@@ -223,6 +243,31 @@ public function findClassMethod(string $className, string $methodName): ?ClassMe
             }
         }
 
+        return $this->getClassMethodOutsideFile($classReflection, $className, $methodName);
+    }
+
+    private function getClassMethodOutsideFile(ClassReflection $classReflection, string $className, string $methodName): ?ClassMethod
+    {
+        if (! $classReflection->hasMethod($methodName)) {
+            return null;
+        }
+
+        $fileName = $classReflection->getFileName();
+        if (! $fileName) {
+            return null;
+        }
+
+        $fileContent = $this->smartFileSystem->readfile($fileName);
+        try {
+            $nodes = $this->parser->parse($fileContent);
+            return $this->getClassMethodFromNodes((array) $nodes, $classReflection, $className, $methodName);
+        } catch (Error $error) {
+            $isNotSyntaxError = ! Strings::contains($error->getMessage(), 'Syntax error');
+            if ($isNotSyntaxError) {
+                throw $error;
+            }
+        }
```

Comment:
Why is this needed? It might hide unwated bugs in broken code

Reply:
It got error on php < 8 on `FluentChainMethodCallToNormalMethodCallRector` tests: 

```bash
1) Rector\Tests\Defluent\Rector\MethodCall\FluentChainMethodCallToNormalMethodCallRector\FluentChainMethodCallToNormalMethodCallRectorTest::test with data set #17 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
59
PhpParser\Error: Syntax error, unexpected T_CONSTANT_ENCAPSED_STRING on line 208
```

ref https://github.com/rectorphp/rector/runs/2454146699#step:5:58

Reply:
What file is that?

Reply:
in these 2 fixture files:

1. https://github.com/rectorphp/rector/blob/main/rules-tests/Defluent/Rector/MethodCall/FluentChainMethodCallToNormalMethodCallRector/Fixture/skip_date_time.php.inc

2. https://github.com/rectorphp/rector/blob/main/rules-tests/Defluent/Rector/MethodCall/FluentChainMethodCallToNormalMethodCallRector/Fixture/skip_date_time_modify_command.php.inc

Reply:
Looking at the fixture files, it seems the erorr is caused by try to fetch to native class DateTime's method, so it may need `isBuiltIn()` check

Reply:
Yes, it needs `isBuiltIn()` check https://github.com/rectorphp/rector/pull/6256/commits/7abde9909df1e8eebe23f8f2b61969d0f6ac8806 🎉